core/storage/statistics/gae_models.py

```diff
@@ -23,11 +23,29 @@
 import json
 import sys
 
+# TODO(#13594): After the domain layer is refactored to be independent of
+# the storage layer, the disable=invalid-import will
+# be removed.
+# The following import is dangerous and should not be generally
+# used. We had to use this ignore as we need to import the domain layer
+# for type-annotation.
+# Here exp_domain is imported outside the 'if MYPY:' block. If it is
+# imported inside the 'if MYPY:' block, during runtime exp_domain
+# won't be accessible.
+from core.domain import exp_domain # pylint: disable=invalid-import
```

Comment:
Wait why cannot we import this in the `if MYPY` block, it is only used in types.

Reply:
If I use the following, then in the runtime python will throw an error that exp_domain is not found.
```
if MYPY:
  from core.domain import exp_domain

def func(x: str) -> exp_domain.Class: ...
```
Thanks!

Reply:
Could you look a this https://stackoverflow.com/a/39757388, I think if we introduce `from __future__ import annotations` and then `if TYPE_CHECKING:` block, it should work.

Reply:
Even without using this import, we can do so by enclosing the type in quotes. Thanks!

Reply:
@vojtechjelinek Just to inform you, following disables will be required for domain imports in `if MYPY` block.
`# isort:skip # pylint: disable=invalid-import,unused-import,ungrouped-imports`
1. `# isort: skip`: Otherwise it tells me to import like this:
 ```
(1)from core.domain import (
(2) exp_domain) # pylint: disable=...
 ```
    This cannot be used here because the pylint disable flag must go in line 1 not in line2, but isort rejects pylint disable in line 1.
2. `invalid-import`: importing domain in storage layer.
3. `unused-import`: pylint thinks the domain modules are unused (their usage is inside quotes like this.  `'exp_domain.Object'`
4. `ungrouped-imports`: these domain imports are not with other imports from `core` module.

Reply:
Fine by me, isort is really annoying with these errors, we should try to find a way where isort doesn't report the longlines (as we already have pylint for that)

Reply:
> Even without using this import, we can do so by enclosing the type in quotes. Thanks!

Did you mean that even with `from __future__ import annotations` we have to enclose it in quotes? Why?

Reply:
@vojtechjelinek I did this with using this import. We have made an issue to decide on this whether we should import this or not. So that's why I have not used the import here and proceeded with the quotes approach. Thanks!

Reply:
I think I used this import somewhere already (while doing the Python 3 migration), but yeah, it is not immediately needed.